### ISSUE TITLE
Overlay issue if other pwnagotchi in range

### ISSUE THREAD

**Issue by mkoterski** [c-nagy/pwnagotchi-display-password-plugin#3]
If there is another pwnagotchi in range, the notification will overlap with what is being displayed by display-password.py.

![image](https://user-images.githubusercontent.com/20810263/107249520-bd7e9d00-6a33-11eb-9c4c-19b108dd766e.png)

According to the [documentation](https://pwnagotchi.ai/plugins/#developing-your-own-plugin) there are the two callbacks that might be of help here

'on_peer_detected
Called when a new peer is detected.

and 

'on_peer_lost
Called when a known peer is lost.

Not sure if I am skilled enough but I might have a look into this if there's interest/time.

**Comment by c-nagy** [comment]
Good catch! I think this can be fixed in a pretty straightforward way. I just wish I had another Pi to test peering and everything with.

If you can put in a fix that would be greatly appreciated. I don't know if the ideal approach would be to have the peer status and cracked passwords alternate every N screen refreshes, or if one should just override the other?

**Comment by NoxiousKarn** [comment]
def __init__(self):
    self.peers_detected = False

def on_loaded(self):
    logging.info("display-password loaded")

def on_ui_setup(self, ui):
    if ui.is_waveshare_v2():
        h_pos = (0, 95)
        v_pos = (180, 61)
    elif ui.is_waveshare_v1():
        h_pos = (0, 95)
        v_pos = (170, 61)
    elif ui.is_waveshare144lcd():
        h_pos = (0, 92)
        v_pos = (78, 67)
    elif ui.is_inky():
        h_pos = (0, 83)
        v_pos = (165, 54)
    elif ui.is_waveshare27inch():
        h_pos = (0, 153)
        v_pos = (216, 122)
    else:
        h_pos = (0, 91)
        v_pos = (180, 61)

    if self.options['orientation'] == "vertical":
        ui.add_element('display-password', LabeledValue(color=BLACK, label='', value='',
                                               position=v_pos,
                                               label_font=fonts.Bold, text_font=fonts.Small))
    else:
        # default to horizontal
        ui.add_element('display-password', LabeledValue(color=BLACK, label='', value='',
                                               position=h_pos,
                                               label_font=fonts.Bold, text_font=fonts.Small))

def on_unload(self, ui):
    with ui._lock:
        ui.remove_element('display-password')

def on_ui_update(self, ui):
    if self.peers_detected:
        # Hide the UI element when peers are detected
        ui.hide_element('display-password')
    else:
        # Display the UI element if no peers are detected
        ui.show_element('display-password')

    last_line = 'tail -n 1 /root/handshakes/wpa-sec.cracked.potfile | awk -F: \'{print $3 " - " $4}\''
    ui.set('display-password', "%s" % (os.popen(last_line).read().rstrip()))

# Called when a new peer is detected
def on_peer_detected(self, agent, peer):
    self.peers_detected = True

# Called when a known peer is lost
def on_peer_lost(self, agent, peer):
    self.peers_detected = False

**Comment by NoxiousKarn** [comment]
[display-password.txt](https://github.com/c-nagy/pwnagotchi-display-password-plugin/files/12544969/display-password.txt)

I think this might work...

**Comment by NoxiousKarn** [comment]
Pull request submitted, sorry it took 3 years.

**Comment by NBalciunas** [comment]
Sorry first time when I actually know how to fix a problem so I don't know how to push my fix into here, but I forked this plugin and made it so that you set the coordinates yourself instead of it detecting your screen. It makes it that you can modify it to be a bit higher or to the side so the texts don't collide.

**Comment by NBalciunas** [comment]
Example image with the exp plugin
![image](https://github.com/c-nagy/pwnagotchi-display-password-plugin/assets/87865572/e5d754eb-cb25-49f0-8e95-eeb719e694bc)

**Comment by dyni00** [comment]
> Example image with the exp plugin ![image](https://github.com/c-nagy/pwnagotchi-display-password-plugin/assets/87865572/e5d754eb-cb25-49f0-8e95-eeb719e694bc)

how do you did it? :)

**Comment by NBalciunas** [comment]
> how do you did it? :)

Hi. If you download [my fork](https://github.com/NBalciunas/pwnagotchi-display-password-plugin) you can set the coordinates for the text manually in the settings!

```
main.plugins.display-password.h_pos_x = 0
main.plugins.display-password.h_pos_y = 91
main.plugins.display-password.v_pos_x = 180
main.plugins.display-password.v_pos_y = 61

```
This is the part you edit to make it work best for yourself. It's part of the .toml file. The settings here were the ones that worked best for me. Also if I remeber correctly you can edit these variables in the pwnagotchi web settings menu.
